qutebrowser/browser/webengine/webenginetab.py

```diff
@@ -246,8 +251,50 @@ def selection(self, html=False):
             raise browsertab.UnsupportedOperationError
         return self._widget.selectedText()
 
+    def _follow_selected_cb(self, js_elem, tab=False, repeat_limit=0):
+        """Callback for javascript which clicks the selected element.
+
+        Used for follow_selected below
+
+        Args:
+            js_elems: The elements serialized from javascript.
+            tab: Open in a new tab or not.
+            repeat_limit: The number of times we have to retry this function."""
+        if js_elem is None:
+            # Try to repeat this request
+            # TODO when on Qt5.9, don't repeat at all, and use selectionChanged
+            # https://bugreports.qt.io/browse/QTBUG-53134
+            if repeat_limit:
+                QTimer.singleShot(
```

Comment:
@The-Compiler To try and work around the race condition, I added a bit of repeating code here. Basically, if we don't find a selection, we rerun the JS 100ms later, and see if it works then. I try this 10 times before giving up completely. If at any point we see something, I click it (below). 

I want to try to run this as infrequently as possible, right now I'm running it if I don't see a selection already (on the python side). Would you be :+1:  on me adding a bit of logic to [AbstractSearch](https://github.com/qutebrowser/qutebrowser/blob/master/qutebrowser/browser/browsertab.py#L152) (or maybe WebEngineSearch) to try and check if we are currently 'searching'? (basically making a boolean true when we run a next_result/prev_result, and clearing it when we run clear).